app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,261 @@ def self.display_methods
   end
 
   menu_section :net
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
+    params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
+
+    @refresh_div = "main_div"
+
+    case params[:pressed]
+    when "security_group_tag"
+      return tag("SecurityGroup")
+    when 'security_group_delete'
+      delete_security_groups
+    when "security_group_edit"
+      checked_security_group_id = get_checked_security_group_id(params)
+      javascript_redirect :action => "edit", :id => checked_security_group_id
+    else
+      if params[:pressed] == "security_group_new"
+        javascript_redirect :action => "new"
+      elsif !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
+        replace_gtl_main_div
+      else
+        render_flash
+      end
+    end
+  end
+
+  def cancel_action(message)
+    session[:edit] = nil
+    @breadcrumbs.pop if @breadcrumbs
+    javascript_redirect :action    => @lastaction,
+                        :id        => @security_group.id,
+                        :display   => session[:security_group_display],
+                        :flash_msg => message
+  end
+
+  def security_group_form_fields
+    assert_privileges("security_group_edit")
+    security_group = find_by_id_filtered(SecurityGroup, params[:id])
+    render :json => {
+      :name              => security_group.name,
+      :description       => security_group.description,
+      :cloud_tenant_name => security_group.cloud_tenant.try(:name),
+    }
+  end
+
+  def create
+    assert_privileges("security_group_new")
+    case params[:button]
+    when "cancel"
+      javascript_redirect :action    => 'show_list',
+                          :flash_msg => _("Add of new Security Group was cancelled by the user")
+    when "add"
+      @security_group = SecurityGroup.new
+      options = form_params
+      ems = ExtManagementSystem.find(options[:ems_id])
+      if SecurityGroup.class_by_ems(ems).supports_create?
+        options.delete(:ems_id)
+        task_id = ems.create_security_group_queue(session[:userid], options)
+
+        add_flash(_("Security Group creation: Task start failed: ID [%{id}]") %
+                  {:id => task_id.to_s}, :error) unless task_id.kind_of?(Fixnum)
```

Comment:
Fixnum should be Integer now (being fixed in https://github.com/ManageIQ/manageiq/pull/13191 for the rest)